tests/reverse/test_catalogue.py

```diff
@@ -953,3 +953,49 @@ def test_product_class_fields_to_update(self):
         prd.refresh_from_db()
         # Product class is updated as it was added in fields_to_update
         self.assertEqual(prd.product_class.requires_shipping, True)
+
+    def test_check_multiple_products_with_same_new_product_class(self):
+        product_resources = [
+            ProductResource(
+                upc="checking",
+                title="Checking",
+                slug="checking",
+                structure=Product.STANDALONE,
+                product_class=ProductClassResource(
+                    name="Better",
+                    slug="better",
+                    requires_shipping=False,
+                    track_stock=True,
+                ),
+            ),
+            ProductResource(
+                upc="testing",
+                title="Testing",
+                slug="testing",
+                structure=Product.STANDALONE,
+                product_class=ProductClassResource(
+                    name="Better",
+                    slug="better",
+                    requires_shipping=False,
+                    track_stock=True,
+                ),
+            ),
+        ]
+        _, errors = products_to_db(product_resources)
+        self.assertEqual(len(errors), 0)
+
+    def test_product_import_with_non_existent_product_class(self):
+        product_resources = [
+            ProductResource(
+                upc="checking",
+                title="Checking",
+                slug="checking",
+                structure=Product.STANDALONE,
+                product_class=ProductClassResource(
+                    slug="better",
```

Comment:
I meant create a case to where you get one product class to create and then another on that does not exist, so also 2 products but one with a new productvlass and one that gives error